src/mbgl/layermanager/fill_layer_factory.cpp

```diff
@@ -27,8 +27,10 @@ std::unique_ptr<style::Layer> FillLayerFactory::createLayer(const std::string& i
 std::unique_ptr<Layout>
 FillLayerFactory::createLayout(const LayoutParameters& parameters,
                                std::unique_ptr<GeometryTileLayer> layer,
-                               const std::vector<const RenderLayer*>& group) noexcept {
-    return std::make_unique<PatternLayout<FillBucket>>(parameters.bucketParameters, group, std::move(layer), parameters.imageDependencies);
+                               const std::vector<Immutable<style::LayerPropertiesWrapper>>& group) noexcept {
+    using namespace style;
+    using LayoutType = PatternLayout<FillBucket, FillPropertiesWrapper, FillPattern, FillPaintProperties::PossiblyEvaluated>;
```

Comment:
`*PropertiesWrapper` already knows about `*PaintProperties::PossiblyEvaluated`, we can infer type from wrapper, will remove one template parameter